gdal/src/main/scala/geotrellis/gdal/GDALReader.scala

```diff
@@ -113,15 +119,19 @@ class GDALReader(val dataset: Dataset) {
         if(cellType.isUnsigned) {
           val ct = cellType match {
             case c: UByteCells with NoDataHandling => c
-            case _ => throw new Exception(s"CellType ${cellType} can't be casted to UByte.")
+            case _ => UByteCellType
```

Comment:
Why do you think the match can fail here? Mb it's better to throwing an exception in a failure case (still wondering how it's even possible rather than user will define his own type, and in this case it's definitely better to throw an exception o:?).

Reply:
Oh, I don't think that match could fail. I'm actually not sure how that can happen with those two `case`s